library_generation/new_client/new-client.py

```diff
@@ -0,0 +1,445 @@
+        "api_description": api_description,
+        "client_documentation": client_documentation,
+        "release_level": release_level,
+        "transport": transport,
+        "language": language,
+        "repo": f"{repo}",
+        "repo_short": f"{language}-{api_shortname}",
+        "distribution_name": distribution_name,
+        "api_id": api_id,
+        "library_type": library_type,
+    }
+    if requires_billing:
+        repo_metadata["requires_billing"] = True
+
+    if rest_docs:
+        repo_metadata["rest_documentation"] = rest_docs
+
+    if rpc_docs:
+        repo_metadata["rpc_documentation"] = rpc_docs
+    # Initialize workdir
+    workdir = Path(f"{sys.path[0]}/../../output/java-{api_shortname}").resolve()
+    if os.path.isdir(workdir):
+        sys.exit(
+            "Couldn't create the module because "
+            f"the module {workdir} already exists. In Java client library "
+            "generation, a new API version of an existing module does not "
+            "require new-client.py invocation. "
+            "See go/yoshi-java-new-client#adding-a-new-service-version-by-owlbot."
+        )
+    print(f"Creating a new module {workdir}")
+    os.makedirs(workdir, exist_ok=False)
+    # write .repo-metadata.json file
+    with open(workdir / ".repo-metadata.json", "w") as fp:
+        json.dump(repo_metadata, fp, indent=2)
+
+    # create owlbot.py
+    templates.render(
+        template_name="owlbot.py.j2",
+        output_name=str(workdir / "owlbot.py"),
+        should_include_templates=True,
+        template_excludes=[],
+    )
+
+    # In monorepo, .OwlBot.yaml needs to be in the directory of the module.
+    owlbot_yaml_location_from_module = ".OwlBot.yaml"
+    # create owlbot config
+    templates.render(
+        template_name="owlbot.yaml.monorepo.j2",
+        output_name=str(workdir / owlbot_yaml_location_from_module),
+        artifact_name=distribution_name_short,
+        proto_path=proto_path,
+        module_name=f"java-{api_shortname}",
+        api_shortname=api_shortname
+    )
+
+    print(f"Pulling proto from {googleapis_url}")
+    output_dir = Path(f"{sys.path[0]}/../../output").resolve()
+    __sparse_clone(
+        remote_url=googleapis_url,
+        dest=output_dir,
+    )
+    # Find a versioned directory within proto_path
+    # We only need to generate one version of the library as OwlBot
+    # will copy other versions from googleapis-gen.
+    version = __find_version(
+        Path(f"{sys.path[0]}/../../output/{proto_path}").resolve()
+    )
+    versioned_proto_path = f"{proto_path}/{version}"
+    print(f"Generating from {versioned_proto_path}")
+    # parse BUILD.bazel in proto_path
+    client_input = parse(
+        build_path=Path(f"{sys.path[0]}/../../output/{versioned_proto_path}")
+        .resolve(),
+        versioned_path=versioned_proto_path,
+    )
+    repo_root_dir = Path(f"{sys.path[0]}/../../").resolve()
+    generator_version = subprocess.check_output(
```

Comment:
The version of generator is now parsed from WORKSPACE. This is a temporary fix so that release manager doesn't need to specify this value.

The plan is to use the version in config file later.